clouddq/templates/dbt/macros/macros.sql

```diff
@@ -34,61 +34,61 @@
   FROM
     zero_record
   LEFT JOIN
     data
   ON
-    zero_record.rule_binding_id = data.rule_binding_id
+    zero_record._internal_rule_binding_id = data._internal_rule_binding_id
 {% endmacro -%}
 
 {% macro validate_complex_rule(rule_id, rule_configs, rule_binding_id, column_name, fully_qualified_table_name, configs) -%}
   SELECT
-    CURRENT_TIMESTAMP() AS execution_ts,
-    '{{ rule_binding_id }}' AS rule_binding_id,
-    '{{ rule_id }}' AS rule_id,
-    '{{ fully_qualified_table_name }}' AS table_id,
-    CAST(NULL AS STRING) AS column_id,
-    NULL AS column_value,
+    CURRENT_TIMESTAMP() AS _internal_execution_ts,
+    '{{ rule_binding_id }}' AS _internal_rule_binding_id,
+    '{{ rule_id }}' AS _internal_rule_id,
+    '{{ fully_qualified_table_name }}' AS _internal_table_id,
+    CAST(NULL AS STRING) AS _internal_column_id,
+    NULL AS _internal_column_value,
 {% if rule_configs.get("dimension") %}
-    '{{ rule_configs.get("dimension") }}' AS dimension,
+    '{{ rule_configs.get("dimension") }}' AS _internal_dimension,
 {% else %}
-    CAST(NULL AS STRING) AS dimension,
+    CAST(NULL AS STRING) AS _internal_dimension,
 {% endif %}
-    CAST(NULL AS BOOLEAN) AS simple_rule_row_is_valid,
-    TRUE AS skip_null_count,
-    custom_sql_statement_validation_errors.complex_rule_validation_errors_count AS complex_rule_validation_errors_count,
+    CAST(NULL AS BOOLEAN) AS _internal_simple_rule_row_is_valid,
+    TRUE AS _internal_skip_null_count,
+    custom_sql_statement_validation_errors._internal_complex_rule_validation_errors_count AS _internal_complex_rule_validation_errors_count,
     CASE
-      WHEN custom_sql_statement_validation_errors.complex_rule_validation_errors_count IS NULL THEN TRUE
-      WHEN custom_sql_statement_validation_errors.complex_rule_validation_errors_count = 0 THEN TRUE
+      WHEN custom_sql_statement_validation_errors._internal_complex_rule_validation_errors_count IS NULL THEN TRUE
+      WHEN custom_sql_statement_validation_errors._internal_complex_rule_validation_errors_count = 0 THEN TRUE
       ELSE FALSE
-    END AS complex_rule_validation_success_flag,
+    END AS _internal_complex_rule_validation_success_flag,
     r"""{{ configs.get(rule_binding_id ~ '_' ~ rule_id ~ '_failed_records_sql_string') }}"""
-    AS failed_records_query,
+    AS _internal_failed_records_query,
   FROM
     zero_record
   LEFT JOIN
     (
       SELECT
          *,
         '{{ rule_binding_id }}' AS _rule_binding_id,
-        COUNT(*) OVER() AS complex_rule_validation_errors_count,
+        COUNT(*) OVER() AS _internal_complex_rule_validation_errors_count,
       FROM (
       {{ rule_configs.get("rule_sql_expr") }}
       ) custom_sql
     ) custom_sql_statement_validation_errors
   ON
-    zero_record.rule_binding_id = custom_sql_statement_validation_errors._rule_binding_id
+    zero_record._internal_rule_binding_id = custom_sql_statement_validation_errors._rule_binding_id
 {% endmacro -%}
 
```

Comment:
@shourya116 Please update the macros for failed_records_query as well to avoid ambiguous errors while running the generated failed_records_query.